docs/deployment/azure/azure-security-best-practices.md

```diff
@@ -0,0 +1,201 @@
+
+// Reference Key Vault in your services
+builder.AddProject<Projects.ApiService>()
+       .WithReference(keyVault);
+```
+
+For more information, see [.NET Aspire Azure Key Vault integration](../../security/azure-security-key-vault-integration.md).
+
+### Configure user-assigned managed identities
+
+For more granular control over permissions, use user-assigned managed identities:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Create or reference a user-assigned managed identity
+var managedIdentity = builder.AddAzureUserAssignedManagedIdentity("app-identity");
+
+// Apply the identity to your container apps
+builder.AddProject<Projects.ApiService>()
+       .WithReference(managedIdentity);
+```
+
+For detailed guidance, see [User-assigned managed identity](../../azure/user-assigned-managed-identity.md).
+
+### Implement network security
+
+**Virtual network integration**: Deploy your Container Apps environment within a custom virtual network for enhanced network isolation:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Configure custom virtual network
+builder.AddAzureContainerAppsEnvironment("aspire-env")
+       .WithVirtualNetwork("custom-vnet");
+```
+
+**Network security groups**: Configure network security groups (NSGs) to control inbound and outbound traffic to your virtual network.
+
+**Private endpoints**: For resources that support it, configure private endpoints to ensure traffic doesn't traverse the public internet.
+
+### Enable comprehensive monitoring
+
+**Application Insights**: Monitor your applications for security anomalies and performance issues:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Add Application Insights
+var appInsights = builder.AddAzureApplicationInsights("monitoring");
+
+builder.AddProject<Projects.ApiService>()
+       .WithReference(appInsights);
+```
+
+For more information, see [Use .NET Aspire with Application Insights](application-insights.md).
+
+**Azure Monitor**: Configure alerts for suspicious activities and resource consumption anomalies.
+
+### Secure data storage
+
+**Encrypt data at rest**: Ensure all data storage resources use encryption at rest. Azure services like Azure SQL Database and Azure Cosmos DB provide this by default.
+
+**Configure backup retention**: Implement appropriate backup and retention policies for your data:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Configure Azure SQL with backup settings
+var sqlServer = builder.AddAzureSqlServer("sql-server")
+                      .WithBackupRetentionDays(30);
+```
+
+**Data classification**: Classify sensitive data and apply appropriate protection measures using Azure Information Protection.
+
+## Production deployment considerations
+
+### Environment separation
+
+Maintain separate environments for development, staging, and production with different security configurations:
+
+```bash
```

Comment:
Anytime you should an `azd` command, the correct language slug is `azdeveloper`.

```suggestion
```azdeveloper
```